### ISSUE TITLE
CMD + A closing window instead of selecting all text

### ISSUE THREAD

**Issue by geoffreyhach** [rbreaves/kinto#788]
**Describe the bug**
The CMD + A shortcut close the current window instead of selecting all text.

**Expected behavior**
I expect it to select all text

**Distro:** Ubuntu 22.04
** Keyboard: ** Logitech mx mechanical mini for mac

**Additional context**
Thanks for the good work !

**Comment by RedBearAK** [comment]
@geoffreyhach 

I’m not the developer but I can explain what’s happening in this case. 

Kinto doesn’t work well with non-US keyboards because there is no mechanism in the keymapper (xkeysnail) to recognize keyboard layout changes. Without rewriting the `key.py` file that defines how the key codes map to actual symbols, it will always act like you’re on a US layout. You can fix individual shortcuts in your config, but it will be pretty confusing. Might work out for you, but only if you never change the keyboard layout. 

I guessed right away that you were on a French layout, since the “A” is where “W” would be on the US layout. 

Basically, software under Kinto’s config operates too close to the hardware and just doesn’t know anything about the keyboard layout/language. It only knows about the hardware key codes, and those don’t change when you change the language or layout. 

I’ve been searching for a solution to this but haven’t found one yet. The xkeysnail fork `keyszer` has a lot of improvements but still suffers from this problem.

**Comment by geoffreyhach** [comment]
@RedBearAK Thanks for you answer ! I'm going to look the resources you gave me.

**Comment by RedBearAK** [comment]
@geoffreyhach 

If you do end up having the patience to modify the `key.py` file to work correctly with the French layout, please post a copy of it here. We may be able to use it in the future.

**Comment by redoxate** [comment]
Hello @RedBearAK , thank you for the refs, I am encountering the same issue, AZERTY layout are ... :-1: 
I am not able to find the key.py file, are you referring to https://github.com/rbreaves/kinto/blob/master/linux/kinto.py

**Comment by RedBearAK** [comment]
@redoxate 

Sorry, the “key.py” file is not part of Kinto, it’s part of the keymapper, “xkeysnail”, and will also be found in the fork of “xkeysnail”, “keyszer”. 

The way Kinto installs the keymapper it will be located with other Python packages in /usr/local/lib/ somewhere, and the exact location will depend on the Python version on your Linux distro. But it would be something like /usr/local/lib/python3.10/site-packages/xkeysnail/.

**Comment by redoxate** [comment]
@rbreaves thanks for the details, I think I fould a less expensive way :
For people encountering this issue, commenting lines containing` K("RC-q") / K("RC-Q") `did the trick for me (Keyboard type: Mac on Ubuntu)

**Comment by RedBearAK** [comment]
@redoxate 

> For people encountering this issue, commenting lines containingK("RC-q") / K("RC-Q")did the trick for me (Keyboard type: Mac on Ubuntu)

That will work for this particular problem, but only if you never need to remap `Cmd+Q` to anything else for a weird application. And there will be numerous other keys on a non-US layout that may act strange if they are remapped by the config. 

I actually tried to fix this with just a `modmap` that would overlay onto a US layout, and that went OK until I realized that some of the keys on non-US keyboards aren't just moved to a different spot, they use a "Shifted" character that you can't transpose onto with a `modmap`. So really the only way this can be entirely fixed is by modifying the key definition file so that it describes the specific layout better. The key definition file doesn't have any "Shifted" characters in it because it only cares about the "base" key symbol that you get when you press the key without `Shift`. Some key codes will need to be moved to a different existing label/variable, but some key codes will need to actually have a new label that wouldn't exist in the key definition for the US layout. 

I believe this can be solved, it just won't exactly be simple.

**Comment by SebCorbin** [comment]
Duplicate of #638 ?

**Comment by Naim-Sirt** [comment]
does anyone found the solution to make works cmd+A as "select all" ?
